api.js

```diff
@@ -1902,4 +1902,49 @@ API.prototype.ignoreApdex = function ignoreApdex() {
   transaction.ignoreApdex = true
 }
 
+/**
+ * Runs a function synchronously within a provided LLM custom attributes context and returns its return value.
+ *
+ * See documentation for newrelic.withLlmCustomAttributes for more information.
+ *
+ * An example of setting a custom attribute:
+ *
+ *    newrelic.withLlmCustomAttributes({'llm.someAttribute': 'someVallue'}, () => {
+ *      return;
+ *    })
+ * @param {Object} context LLM custom attributes context
+ * @param {Function} callback synchronous function called within the context
+ */
+API.prototype.withLlmCustomAttributes = function withLlmCustomAttributes(context, callback) {
+  const metric = this.agent.metrics.getOrCreateMetric(
+    NAMES.SUPPORTABILITY.API + '/withLlmCustomAttributes'
+  )
+  metric.incrementCallCount()
+
+  const transaction = this.agent.tracer.getTransaction()
+  if (!transaction) {
+    logger.warn('withLlmCustomAttributes must be called within the scope of a transaction.')
+    return callback?.()
+  }
+
+  for (const key in context) {
+    if (Object.hasOwn(context, key)) {
+      const value = context[key]
+      if (typeof value === 'object' || typeof value === 'function') {
```

Comment:
Is it possible for `value` to be an array?

Reply:
such a value will have an `object` type so will be dismissed.